shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -223,8 +223,8 @@ final class TsTypegenCodeBuilder {
             toTsType(fieldType)(TypegenContext(fieldType), Some(true))
           case uniqueFieldTypes =>
             // multiple types are intersected hence typegen is done
-            // using intersection precedence however only unique types
-            // are considered for intersection
+            // using precedence of the intersection operator. However
+            // only distinct types are considered for intersection
             val fieldTypesCode = uniqueFieldTypes.toSet[Type].map(fieldType => toTsType(fieldType, false, 1)(TypegenContext(fieldType), Some(true))).toList
             SourceCode.sepBy(fieldTypesCode, SourceCode.ampersand)
```

Comment:
Please remove code+comment the duplication. Having entire sections of code copy-pasted like this is not good practice. In fact the two parts have already started diverging (still using `toSet` here).

Reply:
Added a separate private method for it.